src/terrain/terrain.cpp

```diff
@@ -837,18 +878,18 @@ Terrain::get_path(
         for (Node<const T>* n : adjacent_nodes) {
             // if can stand on the tile    and the tile is not explored
             // get_adjacent should only give open nodes
+
+            // openNodes.remove n
+            n->explore(choice, get_G_cost(*(n->get_tile()), *choice));
```

Comment:
this is wrong. please don't change code if you don't know that it is doing. You are just going to force me to spend time bugfixing in the future.

Reply:
so, I guess you didn't change anything, but because you did, you should have done what was written
`
                // openNodes.remove n
                n->explore(choice, get_G_cost(*(n->get_tile()), *choice));
                // openNodes.add n`

Reply:
it should reorder the queue, I guess one could test if the order actually changed.
https://stackoverflow.com/a/5811888/16735808